lib/jekyll-github-metadata/ghp_metadata_generator.rb

```diff
@@ -8,6 +8,9 @@ class GHPMetadataGenerator < Jekyll::Generator
       def generate(site)
         Jekyll::GitHubMetadata.log :debug, "Initializing..."
 
+        # Fail loudly if credentials are provided but aren't valid.
+        Jekyll::GitHubMetadata.client.check_credentials!
```

Comment:
Won't this be a change in behavior? Right now, I can have an invalid credential, but if I don't use any `site.github` metadata, the drop isn't called, and thus it's okay, right? I don't know that we need to go out of our way to fail a user's build if they're not relying on the plugin for their site to build.